### PR TITLE
Shrimp Mob for Metempsychosis

### PR DESCRIPTION
**Pull request by leonardo-dabepis**
## About the PR

Adds shrimp mob, but only as a rare result of metempsychosis. They have pyrokenisis to make up for their lack in most other things.
## Why / Balance

I got high and thought it'd be funny. This will probably get denied but it's worth the shot.
## Media

image

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- add: Shrimp morphotype*

### CODE REVIEW

**Comment by FluffiestFloof** [comment]
🦐 shrimple as that

**Comment by musicmanvr** [comment]
SHRIMP!!!!

**Comment by rosieposieeee** [comment]
real

**Comment by DebugOk** [comment]
This will absolutely not be merged until the psionics rework is done. Pyrokenisis in its current state is not at all balanced

**Comment by leonardo-dabepis** [comment]
Originally I had tried to give them built in laser carbines, but I couldn't figure it out so I settled for pyrokenisis. I can continue to reduce their power level though. My thought was they need some sort of natural defense since they cannot use weapons or do much of anything aside from pull objects and speak, but I can think of something else.

**Comment by UnicornOnLSD** [comment]
based

File: Resources/Prototypes/DeltaV/Entities/Mobs/NPCs/fun.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
+- type: entity
+  name: space shrimp
+  parent: [ SimpleMobBase, FlyingMobBase, MobCombat ]
+  id: MobSpaceShrimp
+  description: Shrimply cursed...
+  suffix: METEM MACHINE ONLY #for now, only to be used as a very rare result for the metempsychosis machine. If fishing ever gets added, this suffix can be removed.
+  components:
+  - type: CombatMode
```
Comment (DEATHB4DEFEAT) [inline Resources/Prototypes/DeltaV/Entities/Mobs/NPCs/fun.yml]: Doesn't the parented `MobCombat` have this?

File: Resources/Prototypes/DeltaV/Entities/Mobs/NPCs/fun.yml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  - type: Bloodstream
+    bloodMaxVolume: 100
+    bloodReagent: BbqSauce #australia reference
+  - type: InteractionPopup
+    successChance: 0.5
+    interactSuccessString: petting-success-possum
+    interactFailureString: petting-failure-possum
+    interactSuccessSpawn: EffectHearts
+    interactSuccessSound:
+      path: /Audio/Animals/raccoon_chatter.ogg
```
Comment (DEATHB4DEFEAT) [inline Resources/Prototypes/DeltaV/Entities/Mobs/NPCs/fun.yml:50]: Funky sounding shrimp

File: Resources/Textures/DeltaV/Mobs/Animals/shrimp.rsi/meta.json [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+          0.5,
+          0.3
+        ],
+        [
+          0.3,
+          0.4,
+          0.5,
+          0.3
+        ]
+      ]
```
Comment (DEATHB4DEFEAT) [inline Resources/Textures/DeltaV/Mobs/Animals/shrimp.rsi/meta.json]:
```suggestion
      "delays": [
        [ 0.3, 0.4, 0.5, 0.3 ],
        [ 0.3, 0.4, 0.5, 0.3 ],
        [ 0.3, 0.4, 0.5, 0.3 ],
        [ 0.3, 0.4, 0.5, 0.3 ]
      ]
```

File: Resources/Textures/DeltaV/Mobs/Animals/shrimp.rsi/meta.json [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    },
+    {
+      "name": "dead",
+      "delays": [
+        [
+          1
+        ]
+      ]
```
Comment (DEATHB4DEFEAT) [inline Resources/Textures/DeltaV/Mobs/Animals/shrimp.rsi/meta.json]:
```suggestion
```

**Review by DEATHB4DEFEAT — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/DeltaV/Entities/Mobs/NPCs/fun.yml (hunk 2 above)
Comment (leonardo-dabepis) [reply Resources/Prototypes/DeltaV/Entities/Mobs/NPCs/fun.yml:50]: this is intended

File: Resources/Prototypes/DeltaV/Entities/Mobs/NPCs/fun.yml (hunk 1 above)
Comment (leonardo-dabepis) [reply Resources/Prototypes/DeltaV/Entities/Mobs/NPCs/fun.yml]: I don't know, I was high when I made this

**Review by DebugOk — COMMENTED** [review]
If you can do the changes Death requested, and temporarily remove the fire shit until psionics is refactors it should be good

**Review by DEATHB4DEFEAT — DISMISSED** [review]

**Comment by DEATHB4DEFEAT** [comment]
Fix the tests and this looks fine.

**Comment by leonardo-dabepis** [comment]
> Fix the tests and this looks fine.

how do i fix the tests? 🥲

**Comment by Timemaster99** [comment]
> > Fix the tests and this looks fine.
> 
> how do i fix the tests? 🥲

You go down to the failed tests at the bottom of the page and click on "details". Here's the reason it failed

Resources/Textures/DeltaV/Mobs/Animals/shrimp.rsi: Failed to read meta.json: Expecting property name enclosed in double quotes: line 22 column 5 (char 393)

**Comment by leonardo-dabepis** [comment]
wow it was just one comma stopping the whole thing from working 😭

**Comment by IamVelcroboy** [comment]
> Fix the tests and this looks fine.

Can this be merged then?

**Comment by Colin-Tel** [comment]
image
The shrimp stays levitating when it is knocked into crit, which seems a little silly.

**Comment by Colin-Tel** [comment]
Also the body of the shrimp doesn't stop moving when it's dead, which is a little silly too.

**Comment by leonardo-dabepis** [comment]
> image The shrimp stays levitating when it is knocked into crit, which seems a little silly.

they're just really skilled

**Comment by Axionyxx** [comment]
I can't remember if anything else needed to be done for this, should this be reviewed as is?

**Comment by leonardo-dabepis** [comment]
> I can't remember if anything else needed to be done for this, should this be reviewed as is?

It is ready

**Review by Colin-Tel — APPROVED** [review]
えび！🦐

**Pull request merged**